src/NUnitFramework/framework/Constraints/IndexerConstraint.cs

```diff
@@ -59,12 +59,22 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             Guard.ArgumentNotNull(actual, nameof(actual));
 
+            object indexedValue;
             var actualType = actual as Type ?? actual.GetType();
 
-            var getMethod = Reflect.GetDefaultIndexer(actualType, _argumentTypes) ?? throw new ArgumentException($"Default indexer accepting arguments {MsgUtils.FormatCollection(_arguments)} was not found on {actualType}."); ;
+            if (actualType.IsArray || typeof(Array).IsAssignableFrom(actualType))
+            {
+                var array = actual as Array;
+                indexedValue = array?.GetValue((int)_arguments[0]);
```

Comment:
```suggestion
                var array = (Array)actual;
                indexedValue = array.GetValue((int)_arguments[0]);
```
I think it is certain at this point that an item should be an array, so I'd prefer it to fail if we're wrong instead of returning null.

Reply:
Sure - we can do that. It'll fail with a cast exception, so was erring on the side of caution. Point taken that it is unlikely to ever happen given the precondition check :+1:

Reply:
Ah, right, now I remember. Can't do:
```c#
var array = (Array)actual;
```
because:
```
CS0030: Cannot convert type 'TActual' to 'System.Array'
```